src/util/system.cpp

```diff
@@ -243,7 +249,7 @@ namespace {
 fs::path StripRedundantLastElementsOfPath(const fs::path& path)
 {
     auto result = path;
-    while (fs::PathToString(result.filename()) == ".") {
+    while (result.filename().empty() || fs::PathToString(result.filename()) == ".") {
```

Comment:
In commit "refactor: replace boost::filesystem with std::filesystem" (41d7166c8a598b604aad6c6b1d88ad46e23be247)

Note: Just to explain this change, it's needed because `std path class, unlike boost path class treats paths with trailing slashes as having a final `""` component, instead of a final `"."` component.

This change was originally suggested https://github.com/bitcoin/bitcoin/pull/20744#discussion_r731839209, and a test program for this behavior is https://godbolt.org/z/voadYc8M1)

Reply:
(Unfortunately, links like https://github.com/bitcoin/bitcoin/pull/20744#discussion_r731839209 do not work for me. There are probably too much comments in this PR or it's resolved. I don't know.)

Reply:
> (Unfortunately, links like [#20744 (comment)](https://github.com/bitcoin/bitcoin/pull/20744#discussion_r731839209) do not work for me. There are probably too much comments in this PR or it's resolved. I don't know.)

Github struggles with long discusions threads, and these links often they stop working and then start working again. They're the same link used in github notification emails so not special.